DIRHASH_STANDARD.md

```diff
@@ -0,0 +1,203 @@
+
+#### `on_cyclic_link`: `"raise"`
+The the first option is to consider cyclic links an [error condition](#error-conditions) and raise an appropriate exception when detected (preferably before reaching the recursion limit of the language of implementation!).
+
+#### `on_cyclic_link`: `"hash reference"`
+The other option is to replace the dirhash value for the cyclic link with the hash function hexdigest of the relative path from the link to the target. The path is normalized according to the unix standard (with a forward slash `/` separating directories) and without a leading or trailing slash. 
+
+Sometimes multiple links form cycles together. Without loss of generality, cyclic links are defined as the *first occurrence of a link to a directory that has already been visited on the current branch of recursion*. The real path (or inode and device ID) of visited directories, together with the path relative to the Dirhash root, must typically be cached during traversal of the file tree to identify and resolve cyclic links. For further details, see these [examples](#cyclic-links-examples).
+
+
+### Protocol Options
+Name  | Type             | Default             | Description 
+----  | ---------------- | ------------------- | -----------
+entry_properties | Array of Strings | `["name", "data"]` | Which Directory Entry properties to consider. NOTE that `type` is a mandatory property and should not be provided
+on_cyclic_link | String, One of `"raise"` and `"hash reference"` | `"raise"` | how to handle [cyclic links](#cyclic-links).
+
+The Dirhash Protocol is designed so that the same hash should not be obtained with different Protocol Options. Subsequently, when the same hash is obtained one can be sure that the same Protocol Options were used. The options must still be provided when comparing checksums, but this removes the risk of false positives (confirmation of the checksum) due to wrong options used.
+
+
+
+## Error Conditions
+**Directory Not Accessible**: A not accessible (sub)directory results in an error unless excluded by the `match_patterns` filtering option.
+
+**File Not Accessible**: A not accessible file results in an error if the entry property `data` is used unless excluded by the `match_patterns` filtering option.
+
+**Cyclic Symbolic Links**: Presence of cyclic links, with `on_cyclic_links` set to `raise`.
+
+**Directory Empty**: No (non-empty) directory entries to hash in the Dirhash root directory, given provided Filtering Options and `include_empty_dirs` set to `false`.  
```

Comment:
in that case it should be called `allow_empty_dirs` or something similar I think, or might just be to late to think properly :p

Reply:
This is only considering the root directory being empty and is really just a corner case. If `include_empty_dirs=True` then it makes sense to just return the hash of an empty dir descriptor, but if False, then there is really "nothing" to hash.  `allow_empty_dirs` sounds like you would generally fail if an empty directory is encountered at any sub-level.